src/parser/monk/brewmaster/modules/spells/TigerPalm.js

```diff
@@ -44,52 +46,43 @@ class TigerPalm extends Analyzer {
     };
   }
 
-  on_byPlayer_applybuff(event) {
-    const spellId = event.ability.guid;
-    if (SPELLS.BLACKOUT_COMBO_BUFF.id === spellId) {
-      this.bocBuffActive = true;
-    }
+  constructor(options){
+    super(options);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.BLACKOUT_COMBO_BUFF), this.onGainBOC);
+    this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.BLACKOUT_COMBO_BUFF), this.onGainBOC);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.BLACKOUT_COMBO_BUFF), this.onLoseBOC);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.TIGER_PALM), this.onCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.TIGER_PALM), this.onDamage);
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER).spell(SPELLS.TIGER_PALM), this.onDamageTaken);
```

Comment:
This should not have `spell(SPELLS.TIGER_PALM)` on it.
```suggestion
    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this.onDamageTaken);
```

Reply:
Oh oops that's a copy paste error, monk was one of the 3AM classes :^)